### PR TITLE
fix: add search by URI

### PR DESCRIPTION
**Pull request by gabrielfs7**
Allow Generis search by URI

https://oat-sa.atlassian.net/browse/ADF-268

### CODE REVIEW

File: models/classes/search/strategy/GenerisSearch.php [hunk 1]
```diff
@@ -137,4 +151,9 @@ public function supportCustomIndex()
     {
         return false;
     }
+
+    private function isUriSearch(string $queryString): bool
+    {
+        return strpos($queryString, LOCAL_NAMESPACE) === 0;
```
Comment (boajer) [inline models/classes/search/strategy/GenerisSearch.php:157]: I think imported urls may have namespace that differs from `LOCAL_NAMESPACE`

Comment (gabrielfs7) [reply models/classes/search/strategy/GenerisSearch.php:160]: I know, but I collected this change from previous behavior, so there we also had this issue.

https://github.com/oat-sa/tao-core/commit/72b9ac0c875e455637d973186949ece47b0e8e15#diff-25b24518ef6f53d0c34757fa4437edbca32befe763334330ae7f5cd35c7b66a2L86

Comment (boajer) [reply models/classes/search/strategy/GenerisSearch.php]: I don't really like this, but if that is historical behavior we may stick to it unless someone will complain about these edge cases.

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: models/classes/search/GenerisSearchBridge.php [hunk 2]
```diff
@@ -20,17 +20,43 @@
...
             $query->getTerm(),
             $query->getParentClass(),
             $query->getStartRow(),
             $query->getRows()
         );
     }
+
+    private function isUriSearch(string $queryString): bool
+    {
+        return strpos($queryString, LOCAL_NAMESPACE) === 0
```
Comment (ktsirkou) [inline models/classes/search/GenerisSearchBridge.php:59]: I know that this was working like that before
however it is dangerous as the NAMESPACE could be updated - then old URIs will not be reachable. 
or 
the URI provider could be configured with different namespace that the environment NAMESPACE -then **all** URIs will not be reachable. 

reference: generis/uriProvider.conf.php

```
<?php

/**
 * The Uri Provider configuration contains the service to be used
 * to generate unique URI for newly created resources
 *
 * Providers must implement UriProvider
 *
 * Default uri provider is core_kernel_uri_DatabaseSerialUriProvider
 *
 * Alternatives:
 *     core_kernel_uri_AdvKeyValueUriProvider
 *     oat\generis\model\kernel\uri\MicrotimeRandUriProvider
 *     oat\generis\model\kernel\uri\Bin2HexUriProvider
 *
 * @see oat\generis\model\kernel\uri\UriProvider
 */

return new oat\generis\model\kernel\uri\Bin2HexUriProvider(array(
    'namespace' => 'https://sandbox.docker.localhost/ontologies/tao.rdf#'
));

```
We should consider searching a not direct matching for URIs as well

Comment (gabrielfs7) [reply models/classes/search/GenerisSearchBridge.php:59]: That is why we added `filter_var($queryString, FILTER_VALIDATE_URL)` to search for other URIs too.

Comment (ktsirkou) [reply models/classes/search/GenerisSearchBridge.php:59]: My bad, I didn't see the resolved conversation. Nice catch and improvement over the previous URI search

Comment (gabrielfs7) [reply models/classes/search/GenerisSearchBridge.php:59]: No problem :) It was a good point indeed

**Review by habemuscode — APPROVED** [review]
LGTM

**Review by bartlomiejmarszal — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
